doc/develop/test/twister.rst

```diff
@@ -408,6 +408,16 @@ platform_allow: <list of platforms>
     constraints, this option should only be used if the test or sample can
     only be run on the allowed platform and nothing else.
 
+    .. warning::
+
+      Some platform enables the testing for themselves by default, the test scenario **will** be run
```

Comment:
Boards enable testing in their board definition file (boardname.yaml), using the `testing` keyword. All tests **will** be run for those platforms if the test doesn't specify a list of `allowed_platforms` in the `testcase.yaml`.